src/lib/utilities.ts

```diff
@@ -61,146 +61,40 @@ function toPascalCase(value: string): string {
-  const charactersMatch = value.match(/^[a-zA-Z0-9.\-_]+$/i) !== null;
-  const isValidArgName = isString && charactersMatch;
-  let resultMessage;
-
-  if (isNull) {
-    resultMessage = `${CLI_STATE.Error} A page name is required`;
-  } else if (!charactersMatch) {
-    resultMessage = `${CLI_STATE.Error} Use letters, numbers and '-' for page names (e.g. page-name)`;
-  }
-
-  return isValidArgName ? true : resultMessage;
-}
-
-/**
- * Description: determine if string is valid service name
- * @param {string} value - a string value
- * @returns {any} -
- */
-function validateServiceName(value: any) {
-  const isString = typeof value === 'string';
-  const isNull = value === null || value.length === 0;
-  // characters in value are limited to alphanumeric characters and hyphens or underscores
-  const charactersMatch = value.match(/^[a-zA-Z0-9.\-_]+$/i) !== null;
-  const isValidArgName = isString && charactersMatch;
-  let resultMessage;
-
-  if (isNull) {
-    resultMessage = `${CLI_STATE.Error} A service name is required`;
-  } else if (!charactersMatch) {
-    resultMessage = `${CLI_STATE.Error} Use letters, numbers and '-' for service names (e.g. service-name)`;
-  }
-
-  return isValidArgName ? true : resultMessage;
-}
-
-/**
- * Description: determine if string is valid store module name
- * @param {string | null} value - a string value
+ * @param {string} featureName - the name of the feature whose name is being validated
+ * @param {string} exampleName - an example of a valid name
  * @returns {any} -
  */
-function validateStoreModuleName(value: any) {
-  const isString = typeof value === 'string';
-  const isNull = value === null || value.length === 0;
-  // characters in value are limited to alphanumeric characters and hyphens or underscores
-  const charactersMatch = value.match(/^[a-zA-Z0-9.\-_]+$/i) !== null;
-  const isValidArgName = isString && charactersMatch;
-  let resultMessage;
-
-  if (isNull) {
-    resultMessage = `${CLI_STATE.Error} A store module name is required`;
-  } else if (!charactersMatch) {
-    resultMessage = `${CLI_STATE.Error} Use letters, numbers and '-' for store module names (e.g. auth-store)`;
-  }
+function validateEnteredName(featureName: string, exampleName = '') {
+  return (value: any) => {
+    const isString = typeof value === 'string';
+    const isNull = value === null || value.length === 0;
+    // characters in value are limited to alphanumeric characters and hyphens or underscores
+    const charactersMatch = value.match(/^[a-zA-Z0-9.\-_]+$/i) !== null;
+    const isValidName = isString && charactersMatch;
+    let resultMessage;
+
+    if (isNull) {
+      resultMessage = `${CLI_STATE.Error} A ${featureName} name is required`;
+    } else if (!charactersMatch) {
+      resultMessage = `${CLI_STATE.Error} Use letters, numbers and '-' for ${featureName} names (e.g. ${exampleName ? exampleName : `my-${featureName}-name`})`;
+    }
```

Comment:
I think this optimization makes sense since we have a standardized naming convention that we expect. In future if any name validation deviates from this Regex then we will need to account for that in this method.